opentracing-observer/src/main/java/io/opentracing/contrib/observer/SpanBuilderObserver.java

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.observer;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+/**
+ * This interface represents an observer used to receive notifications related to a {@link SpanBuilder}.
+ *
+ */
+public interface SpanBuilderObserver {
+
+    /**
+     * Notifies observer that the named tag has been set/changed.
+     *
+     * @param name The tag name
+     * @param value The tag value
+     */
+    void onWithTag(String name, Object value);
```

Comment:
Do you mean have one interface `SpanObserver` that is used by the `SpanBuilder` to notify, so it would potentially call `onSetTag` and `onAddReference` before `onStart`, and then afterwards potentially have `onSetTag` called other times before finally `onFinish`?